// engage-call-api-js-server-sdk/lib/services/MultipartyConferenceCallService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultipartyConferenceCallService = void 0;
const request_1 = require("../core/request");
class MultipartyConferenceCallService {
    /**
     * Get all Rooms
     * Get details of all conference rooms based on specified filter criteria. A list of room detail records will be returned in response.<br />
     * @param acId Account Id of the tenant who is trying to perform the operation
     * @param roomName (Optional Parameter) This parameter looks for an exatact match in DB with the given value, and if room exits then record will be given, else empty list will be given as response.
     * @param createdAt (Optional Parameter) Room Creation time.
     * @param closedAt (Optional Parameter) Room Closing time.
     * @param status (Optional Parameter) Specifies the current room status
     * @param roomType (Optional Parameter) Specifies the Room Type
     * @param limit (Optional Parameter) Integer value specifing the limit of rooms to be returned in response.
     * @param offset (Optional Parameter) Integer value specifing the offset to fetch in the paginated room list.
     * @returns RoomDetailRecordItemList successful operation
     * @throws ApiError
     */
    static getMultipleRoomRecords(acId, roomName, createdAt, closedAt, status, roomType, limit, offset) {
        return (0, request_1.request)({
            method: 'GET',
            path: `/v1/accounts/${acId}/room`,
            query: {
                'roomName': roomName,
                'CreatedAt': createdAt,
                'closedAt': closedAt,
                'status': status,
                'roomType': roomType,
                'limit': limit,
                'offset': offset,
            },
            errors: {
                400: `Bad Request`,
                404: `Not Found`,
                408: `Request Timeout`,
            },
        });
    }
    /**
     * Create Room
     * Create a conference room which allows multiple users to join.
     * @param acId Account Id of the tenant who is trying to perform the operation
     * @param requestBody
     * @returns RoomDetailRecord successful operation
     * @throws ApiError
     */
    static createRoom(acId, requestBody) {
        return (0, request_1.request)({
            method: 'POST',
            path: `/v1/accounts/${acId}/room`,
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                400: `Bad Request`,
                404: `Not Found`,
                408: `Request Timeout`,
            },
        });
    }
    /**
     * Get Room
     * Get details of specific room by providing it's room_ID
     * @param acId Account Id of the tenant who is trying to perform the operation
     * @param rmId Room ID, for which user wants to see the details
     * @returns RoomDetailRecord successful operation
     * @throws ApiError
     */
    static getRoomRecordByRoomId(acId, rmId) {
        return (0, request_1.request)({
            method: 'GET',
            path: `/v1/accounts/${acId}/room/${rmId}`,
            errors: {
                400: `Bad Request`,
                404: `Not Found`,
                408: `Request Timeout`,
            },
        });
    }
    /**
     * Delete room
     * Close an existing room. This API acts as a soft delete, which changes the room status to 'CLOSED' and makes the room unavailable for usage. The room record will still be available with a closed status.
     * @param acId Account Id of the tenant who is trying to perform the operation
     * @param rmId Room ID, for which user wants to perform the opration.
     * @returns RoomDetailRecord successful operation
     * @throws ApiError
     */
    static deleteRoom(acId, rmId) {
        return (0, request_1.request)({
            method: 'DELETE',
            path: `/v1/accounts/${acId}/room/${rmId}`,
            errors: {
                400: `Bad Request`,
                408: `Request Timeout`,
            },
        });
    }
    /**
     * Join Room
     * Join to a specific conference room
     * @param acId Account Id of the tenant who is trying to perform the operation
     * @param rmId Room ID, for which user wants to perform the join operation
     * @param requestBody
     * @returns JoinResponse successful operation
     * @throws ApiError
     */
    static joinRoom(acId, rmId, requestBody) {
        return (0, request_1.request)({
            method: 'POST',
            path: `/v1/accounts/${acId}/room/${rmId}/join`,
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                400: `Bad Request`,
                408: `Request Timeout`,
            },
        });
    }
    /**
     * Play Room
     * Play announcement in the Room.
     * @param acId Account Id of the tenant who is trying to perform the operation
     * @param rmId Room ID, for which user wants to perform the play operation
     * @param requestBody
     * @returns ServerResponse successful operation
     * @throws ApiError
     */
    static playAnnouncementInRoom(acId, rmId, requestBody) {
        return (0, request_1.request)({
            method: 'POST',
            path: `/v1/accounts/${acId}/room/${rmId}/play`,
            body: requestBody,
            mediaType: 'application/json',
            errors: {
                400: `Bad Request`,
                408: `Request Timeout`,
            },
        });
    }
}
exports.MultipartyConferenceCallService = MultipartyConferenceCallService;
