// engage-call-api-js-server-sdk/lib/models/RoomBase.js
"use strict";
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoomBase = void 0;
var RoomBase;
(function (RoomBase) {
    /**
     * Optional. This parameter specifies the room type in one of the following values: Audio, Video-VAS, Video-CPV-6-split, Video-CPV-4-split, Video-CPV-ASL, Video-CPV-DHL, Video-CPV-auto-split. Default value should be Audio, setting this field as Video will enable the video in room calls.
     */
    let RoomType;
    (function (RoomType) {
        RoomType["AUDIO"] = "Audio";
        RoomType["VIDEO_VAS"] = "Video-VAS";
        RoomType["VIDEO_CPV_6_SPLIT"] = "Video-CPV-6-split";
        RoomType["VIDEO_CPV_4_SPLIT"] = "Video-CPV-4-split";
        RoomType["VIDEO_CPV_ASL"] = "Video-CPV-ASL";
        RoomType["VIDEO_CPV_DHL"] = "Video-CPV-DHL";
        RoomType["VIDEO_CPV_AUTO_SPLIT"] = "Video-CPV-auto-split";
    })(RoomType = RoomBase.RoomType || (RoomBase.RoomType = {}));
    /**
     * Optional. Default: No. Set to Yes if PSTN Dial-In required
     */
    let GenerateDialInCode;
    (function (GenerateDialInCode) {
        GenerateDialInCode["YES"] = "Yes";
        GenerateDialInCode["NO"] = "No";
    })(GenerateDialInCode = RoomBase.GenerateDialInCode || (RoomBase.GenerateDialInCode = {}));
    /**
     * Optional. This parameter specifies which HTTP method should be used to request the StatsCallback URL (if specified) when sending the call status event. Valid parameters are GET and POST. Default value is POST.
     */
    let Method;
    (function (Method) {
        Method["GET"] = "GET";
        Method["POST"] = "POST";
    })(Method = RoomBase.Method || (RoomBase.Method = {}));
})(RoomBase = exports.RoomBase || (exports.RoomBase = {}));
